File: UNIT 05/SOURCE CODE/JAVA/ComplexNumbersExample.java

package com.dube.csc213;

/**
 *
 * @author Enock Lukhetfo Dube  <eldube@uniswa.sz>
 */
public class ComplexNumbersExample {

   public  static class  Complex {

	private float x;     //real part
	private float y;     //imaginary part

        public Complex() {}
	public Complex(float xp, float yp) { x = xp; y = yp; }
	public float Re() { return x; }
	public float Im() { return y; }


	//Java does not support operator overloading 
        //use methods instead
	public Complex add  (Complex Z)
	{
		x = x + Z.Re();
		y = y + Z.Im();
		return this;
	}

	public Complex multiply (Complex Z)
	{
		float temp = x;
		x = x * Z.Re() - y * Z.Im();
		y = temp * Z.Im() + y * Z.Re();
		return this;
	}
        
        //Java uses print/println method to write output
        // over-ride toString() method to use print/println for Complex numbers

        @Override
        public String toString() {
            return x + " + " + y + 'i';
        }   
     }
   
    public static void main(String[] args) {
        Complex Z1 = new Complex (5,10);
        Complex Z2 = new Complex(4,5);

        System.out.println(Z1);
        System.out.println(Z2);
        
        Z1.add(Z2);
        System.out.println(Z1);
        
        Complex Z3 = new Complex (5,10);
        Z3.multiply(Z2);
        System.out.println(Z3);    
    }
    
}

File: UNIT 05/SOURCE CODE/CSHARP/ComplexNumbersExample.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ComplexNumbersExample
{
    class ComplexNumbersExample
    {

        public  class Complex
        {

            private float x;     //real part
            private float y;     //imaginary part

            public Complex() { }
            public Complex(float xp, float yp) { x = xp; y = yp; }
            public float Re() { return x; }
            public float Im() { return y; }


            //overload addition operator
            public static Complex operator +(Complex Z1, Complex Z2)
            {
                Complex Z = new Complex();
                Z.x = Z1.Re() + Z2.Re();
                Z.y= Z1.Im() + Z2.Im();
                return Z;
            }


            //overload multiplication operator
            public static Complex operator *(Complex Z1, Complex Z2)
            {
                Complex Z = new Complex();
                Z.x = Z1.Re() * Z2.Re() - Z1.Im() * Z2.Im();
                Z.y = Z1.Re() * Z2.Im() + Z1.Im() * Z2.Re();
                return Z;
            }
           

            //C# uses Write/Writeln method to write output
            //over-ride ToString() method to use Write/Writeln for Complex numbers

            public override string ToString()
            {
                return x + " + " + y + 'i';
            }

            
        }
        static void Main(string[] args)
        {
            Complex Z1 = new Complex(5, 10);
            Complex Z2 = new Complex(4, 5);

            Console.WriteLine(Z1);
            Console.WriteLine(Z2);

            Z1 = Z1 + Z2;
            Console.WriteLine(Z1);

            Complex Z3 = new Complex(5, 10);
            Z3 = Z3*Z2;
            Console.WriteLine(Z3);

            Console.ReadKey();

        }
    }
}
